cocos/core/memop/recycle-pool.ts

```diff
@@ -112,6 +111,10 @@ export class RecyclePool<T = any> extends ScalableContainer {
         return this._data[this._count++];
     }
 
+    /**
+     * @en Destroy the object pool. Should resize again if want to reuse the recycle pool.
+     * @zh 销毁对象池。后续如果还想使用的话，需要先扩容。
```

Comment:
I don't think we should encourage reuse a destroyed object, for example, a destroyed node can't be used under any circumstances

Reply:
I don't know the original purpose. Currently can reuse it. So i just mentioned it here. Of course, can remove it if don't encourage it.

Reply:
Fixed. Mention that, don't use it again after it is destroyed.